pkg/kubelet/container/helpers.go

```diff
@@ -45,6 +45,7 @@ type HandlerRunner interface {
 // RuntimeHelper wraps kubelet to make container runtime
 // able to get necessary informations like the RunContainerOptions, DNS settings, Host IP.
 type RuntimeHelper interface {
+	GenerateRunContainerEnvs(pod *v1.Pod, container *v1.Container, podIP string) (contEnvs []EnvVar, err error)
```

Comment:
This is only called in one place inside kubelet_pods.go.  Does it need to be exported as a new interface?

Reply:
@msau42 Probably not now as it is being extracted away from the container runtime. Will look tomorrow and clean up last bits.

Reply:
@msau42 Addressed the 3 cleanup comments